internal/services/arckubernetes/arc_kubernetes_cluster_extension_resource.go

```diff
@@ -0,0 +1,358 @@
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r ArcKubernetesClusterExtensionResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ArcKubernetes.ExtensionsClient
+
+			id, err := extensions.ParseExtensionID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := ArcKubernetesClusterExtensionModel{
+				Name:      id.ExtensionName,
+				ClusterID: arckubernetes.NewConnectedClusterID(metadata.Client.Account.SubscriptionId, id.ResourceGroupName, id.ClusterName).ID(),
+			}
+
+			if model := resp.Model; model != nil {
+				if err = metadata.ResourceData.Set("identity", identity.FlattenSystemAssigned(model.Identity)); err != nil {
+					return fmt.Errorf("setting `identity`: %+v", err)
+				}
+
+				if properties := model.Properties; properties != nil {
+					var originalModel ArcKubernetesClusterExtensionModel
+					if err := metadata.Decode(&originalModel); err != nil {
+						return fmt.Errorf("decoding: %+v", err)
+					}
+
+					state.ConfigurationProtectedSettings = originalModel.ConfigurationProtectedSettings
+
+					if properties.ConfigurationSettings != nil {
+						state.ConfigurationSettings = *properties.ConfigurationSettings
+					}
+
+					if properties.CurrentVersion != nil {
+						state.CurrentVersion = *properties.CurrentVersion
+					}
+
+					if properties.ExtensionType != nil {
+						state.ExtensionType = *properties.ExtensionType
+					}
+
+					if properties.ReleaseTrain != nil {
+						state.ReleaseTrain = *properties.ReleaseTrain
+					}
+
+					if properties.Scope != nil {
+						if properties.Scope.Cluster != nil && properties.Scope.Cluster.ReleaseNamespace != nil {
+							state.ReleaseNamespace = *properties.Scope.Cluster.ReleaseNamespace
+						}
+
+						if properties.Scope.Namespace != nil && properties.Scope.Namespace.TargetNamespace != nil {
+							state.TargetNamespace = *properties.Scope.Namespace.TargetNamespace
+						}
+					}
+
+					if properties.Version != nil {
+						state.Version = *properties.Version
+					}
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r ArcKubernetesClusterExtensionResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ArcKubernetes.ExtensionsClient
+
+			id, err := extensions.ParseExtensionID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id, extensions.DeleteOperationOptions{}); err != nil {
```

Comment:
For future reference if there's an operation method that takes options we should always use the function [`DefaultSomeOperationOptions()`](https://github.com/hashicorp/go-azure-sdk/blob/main/resource-manager/kubernetesconfiguration/2022-11-01/extensions/method_delete_autorest.go#L25) since they should contain any default options that are defined in the swagger.
```suggestion
			if err := client.DeleteThenPoll(ctx, *id, extensions.DefaultDeleteOperationOptions()); err != nil {
```